src/main/java/edu/harvard/iq/dataverse/PermissionServiceBean.java

```diff
@@ -184,7 +187,7 @@ public RequestPermissionQuery on(DvObject dvo) {
      * @param d the object on which the permissions will be queried
      * @return A permission query
      */
-    public RequestPermissionQuery on(DvObject d) {
+    public RequestPermissionQuery on( @NotNull DvObject d) {
         if (d == null) {
```

Comment:
`d` isn't null. that's why to `NotNull` it

Reply:
Ideally, yes. But as far as I can tell, you still need to do runtime validation in regular Java (i.e. not using Lombok et al). See https://blogs.oracle.com/java-platform-group/java-8s-new-type-annotations (scroll down to the section "Optional Type Annotations are not a substitute for runtime validation").

It may generate compiler warnings (https://www.jetbrains.com/help/idea/nullable-and-notnull-annotations.html), but as long as there are no errors from standard `javac`, we have to do the check.

That said, I'll be very happy to see an official document saying we can remove that check :-)

Reply:
It does say that "Your code should still perform runtime validation", but on the other hand, maven and any ide will not allow it to compile, at which point I feel like we should not have runtime checking for something we already check at compile time. It would be like testing isinstance every time we use generics because java's generics are broken

Reply:
Dunno. This compiles on my machine with warning only:

    public static void main(String[] args) {
        new PermissionServiceBean().on(null);
    }

(leaving the Java generics part for the next time we meet 😄)